### PR TITLE
fixed bug 144 and 145 and refactored documentation for nimble info module

### PR DESCRIPTION
**Pull request by ar-india**
https://nimblejira.nimblestorage.com/browse/SA-144

https://nimblejira.nimblestorage.com/browse/SA-145

### CODE REVIEW

File: ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_info.py [hunk 1]
```diff
@@ -37,53 +37,21 @@
...
-      - Except for "all", "minimum" and "config" subsets, each subset supports four types of subset options.
-
-      - fields - A list representing attributes that are displayed for a given subset.
-      - limit - An integer value that represents how many latest items to show for a given subset.
-      - detail - A bool flag that when set to 'True' fetches everything for a given subset. The default is 'True'.
-      - query - A Key-Value pair used to perform a query.
-
-      - Subset 'all' just supports "limit" and "detail" as subset options.
-      - Subset 'config' and 'minimum' does not support any subset options.
+      - When supplied, this argument will define the information to be collected. Possible values for this include "all" "minimum" "config"
```
Comment (ar-india) [inline ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_info.py:40]: just refactored as per michael suggestion

File: ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_info.py [hunk 2]
```diff
@@ -875,8 +851,9 @@ def fetch_snapshots_for_all_subset(subset, client_obj):
                 if subset['limit'] is not None and total_snap.__len__() >= subset['limit']:
                     total_snap = total_snap[0:subset['limit']]
                     break
-        result['snapshots'] = generate_dict('snapshots', total_snap)['snapshots']
-        return result
+        if total_snap.__len__() > 0:
+            result['snapshots'] = generate_dict('snapshots', total_snap)['snapshots']
+    return result
```
Comment (ar-india) [inline ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_info.py:856]: caused bug 144.. return result went inside if statement

File: ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_info.py [hunk 3]
```diff
@@ -875,8 +851,9 @@ def fetch_snapshots_for_all_subset(subset, client_obj):
                 if subset['limit'] is not None and total_snap.__len__() >= subset['limit']:
                     total_snap = total_snap[0:subset['limit']]
                     break
-        result['snapshots'] = generate_dict('snapshots', total_snap)['snapshots']
-        return result
+        if total_snap.__len__() > 0:
```
Comment (ar-india) [inline ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_info.py:854]: this is fix for 145.. there can be a case that a volume has no snapshots at all.. hence no point in calling generate_dict in that case

**Review by suneeth51 — APPROVED** [review]
lgtm...

File: ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_info.py [hunk 4]
```diff
@@ -637,7 +606,7 @@ def is_subset_already_added(key, valid_subset_list):
     return False
 
 
-def handle_all_subset(info_subset, valid_subset_list, subset_options, detail):
+def handle_all_subset(info_subset, valid_subset_list, subset_options):
```
Comment (ar-india) [inline ansible_collection/hpe/nimble/plugins/modules/hpe_nimble_info.py:609]: we are not using detail flag anymore. this should have ben removed long back when i first introduced this function.. just a code cleanup

**Pull request merged**